test/jackdaw/word_count_test.clj

```diff
@@ -0,0 +1,160 @@
+(ns jackdaw.word-count-test
+  (:require [clojure.test :refer :all]
+            [jackdaw.streams :as k]
+            [jackdaw.test :as jd.test]
+            [jackdaw.test.fixtures :as fix])
+  (:import org.apache.kafka.common.serialization.Serdes))
+
+;;; Example of using the test-machine ;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
+;;
+;; Here we take the word-count example from confluent's examples repo
+;;
+;;   https://github.com/confluentinc/kafka-streams-examples
+;;
+;; and provide an implementation in Clojure, and test it using the
+;; test-machine.
+;;
+;; First the app. All pretty simple stuff.
+;;
+;;   - Parse each line into a list of words
+;;   - Group By each unique word
+;;   - Count the records in each group
+;;   - Write the counts to the output topic
+
+(defn parse-line
+  [line]
+  (let [line (-> (.toLowerCase line (java.util.Locale/getDefault)))]
+    (->> (.split line "\\W+")
+         (into []))))
+
+(defn word-count
+  [in out]
```

Comment:
If we switch to a "topic metadata" format like we use in the [serdes example](https://github.com/FundingCircle/jackdaw/blob/master/examples/serdes/src/serdes.clj#L129-L159), this the function could be made to take one argument.